packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -4,14 +4,81 @@ const chalk = require(`chalk`)
 const { formatPluginOptionsForCLI } = require(`./plugin-options`)
 const { CODES } = require(`./report`)
 
+/**
+ * Generate a user friendly error message.
+ *
+ * Contentful's API has its own error message structure, which might change depending of internal server or authentification errors.
+ *
+ * Additionally the SDK strips the error object, sometimes:
+ * https://github.com/contentful/contentful.js/blob/b67b77ac8c919c4ec39203f8cac2043854ab0014/lib/create-contentful-api.js#L89-L99
+ *
+ * This code tries to work around this.
+ */
+const createContentfulErrorMessage = e => {
+  // If we got a response, it is very likely that it is a Contentful API error.
+  if (e.response) {
+    let parsedContentfulErrorData = {}
```

Comment:
we shouldn't set empty object here, null is fine
```suggestion
    let parsedContentfulErrorData = null
```